apps/web/src/app/(private)/_data-access/index.ts

```diff
@@ -54,20 +55,28 @@ export const getDocumentByUuid = cache(
     documentUuid: string
     commit: Commit
   }) => {
-    const workspace = await findWorkspaceFromCommit(commit)
-    const scope = new DocumentVersionsRepository(workspace!.id)
+    const { workspace } = await getCurrentUser()
+    const scope = new DocumentVersionsRepository(workspace.id)
     const result = await scope.getDocumentAtCommit({ documentUuid, commit })
+    if (result.error) {
+      const error = result.error
+      if (error instanceof NotFoundError) {
+        return notFound()
```

Comment:
When is a not found we don't raise the error. We use NextJS `notFound()` method that display the `not-found.tsx` page